### PR TITLE
Fixes: "Axes2D flickering #1341"

### PR DESCRIPTION
**Pull request by nilscb**

### CODE REVIEW

File: react/package.json [hunk 1]
```diff
@@ -77,6 +77,7 @@
         "deck.gl": "^8.8.11",
         "deep-equal": "^2.0.5",
         "fast-json-patch": "^3.0.0-1",
+        "gl-matrix": "^3.4.3",
```
Comment (hkfb) [inline react/package.json:80]: Should also push `package-lock.json` so that we get this package when we run `npm ci`

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/layers/axes2d/axes2DLayer.ts [hunk 2]
```diff
@@ -1,134 +1,256 @@
...
 
-        const screen_from = this.context.viewport.project(d.from);
-        const screen_to = this.context.viewport.project(d.to);
-        const is_labels = d.label !== "X" && d.label !== "Y" && d.label !== "Z"; // labels on axis or XYZ annotations
-        if (is_labels) {
-            if (screen_from[0] < screen_to[0]) {
-                return "start";
-            }
-        }
+        const promise = load("font-atlas.png", ImageLoader, {
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/axes2d/axes2DLayer.ts:124]: I don't think this will work in an npm package. I think you should put the image in a lib directory and use an import statement.

**Review by hkfb — CHANGES_REQUESTED** [review]

**Review by hkfb — CHANGES_REQUESTED** [review]
Can we import the font atlas png using webpack instead? Eg. https://webpack.js.org/guides/asset-management/#loading-images

**Comment by hkfb** [comment]
I think we should test this deployed before merging (in order to test the image file packaging). Maybe @t0oF-azpn could help?

File: react/src/lib/components/DeckGLMap/layers/axes2d/axes2DLayer.stories.tsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ export default {
 } as ComponentMeta<typeof DeckGLMap>;
 
 const layerProps = {
-    marginH: 80, // Horizontal margin (in pixels)
+    marginH: 100, // Horizontal margin (in pixels)
```
Comment (VincentNevermore) [inline react/src/lib/components/DeckGLMap/layers/axes2d/axes2DLayer.stories.tsx:11]: I think we can revert this change here or is there a specific reason that we want to use 100 instead of 80 ?

Comment (VincentNevermore) [inline react/src/lib/components/DeckGLMap/layers/axes2d/axes2DLayer.stories.tsx:11]: I think we should revert it here or is there a specific reason we use 100 ?

**Review by VincentNevermore — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/layers/axes2d/axes2DLayer.stories.tsx (hunk 3 above)
Comment (nilscb) [reply react/src/lib/components/DeckGLMap/layers/axes2d/axes2DLayer.stories.tsx:11]: Horisontal margin needs to be a bit less than the vertical one so I think these are sensible default values,

**Comment by nilscb** [comment]
This is now tested by  @t0oF-azpn and found to be working.

**Review by VincentNevermore — APPROVED** [review]

File: react/package.json [hunk 4]
```diff
@@ -70,13 +70,15 @@
         "@vivaxy/png": "^1.3.0",
         "@webviz/core-components": "^0.6.0",
         "ajv": "^7.2.1",
+        "buffer": "^6.0.3",
```
Comment (hkfb) [inline react/package.json:73]: What is this used for? It seems to work fine without it...

**Review by hkfb — CHANGES_REQUESTED** [review]

**Review by hkfb — APPROVED** [review]
